documentation/modules/exploit/windows/fileformat/device_viewer.md

```diff
@@ -0,0 +1,115 @@
+## Vulnerable Application
+
+This module exploits a SEH stack buffer overflow in DeviceViewer version 3.10.12.0. By creating a specially crafted file and copying its content in the "User" login field, an attacker will be able to gain arbitrary code execution.
+
+Link to vulnerable software [EDB](https://www.exploit-db.com/apps/4d10486a079bd1f1864c30e86cd2aa80-DeviceViewer.exe) - [VoidSec](https://github.com/VoidSec/Exploit-Development/blob/master/windows/x86/local/DeviceViewer_v.3.10.12.0/DeviceViewer.exe)
+
+## Test Results
+  * Windows XP SP3 Pro x64 (DEP & ASLR disabled) - trigger
```

Comment:
I'm confused by this section. First off, you should have a `## Verification Steps` section here, and I'm not seeing one. Take a look at https://github.com/rapid7/metasploit-framework/pull/13200/files and the file `documentation/modules/exploit/windows/local/ntusermndragover.md` for a look at how one of these files should be formatted.

More importantly though, I am very confused why you mention that DEP and ASLR was disabled for the older versions but not for the newer one, particularly given your exploit description that this exploit bypasses ASLR and DEP and that the exploit description, as it stands at the moment, seems to imply that you can bypass ASLR and DEP on all targets without having to disable it.

Can you please clarify this ASLR + DEP requirement for all targets more? Is it that the program doesn't have ASLR or DEP enabled? That Windows 10 forces these settings? That you manually changed the settings? Its not clear from this description.

As this section will need to be removed anyway, please just type your response in a comment below.

Reply:
I've double checked:
  * Windows XP Pro x86 v.5.1.2600 SP 3 Build 2600: ASLR and DEP are disabled on default configuration
  * Windows 7 Pro x86 v.6.1.7601 SP 1 Build 7601: Windows enforce ASLR and DEP automatically on default configuration
  * Windows 10 Pro x64 v.1909 Build 18363.720: Windows enforce ASLR and DEP automatically on default configuration

I was probably testing it on a modified image that I made for testing purpose. I'll rework the exploit to reflect this changes and re-check all the offsets and ROP.

Reply:
Ok Windows XP sounds fine and I was half expecting that. That being said, as you mentioned Windows 7 should probably have the ASLR and DEP bypass if that is on in its default configuration.

Appreciate you posting this though, might be useful to note that ASLR and DEP are disabled on the Windows XP install by default in the documentation.

Reply:
Can confirm that Windows 7 now has a ROP chain along with Windows 10. Will need to set up a Windows XP SP3 machine to confirm the Windows XP settings though, but rest looks good. Marking as resolved for now.